### ISSUE TITLE
Add a `<informative>` macro element

### ISSUE THREAD

**Issue by tabatkins** [tabatkins/bikeshed#954]
Marking things as informative is a little bit annoying now, it could use a shorthand element to make it easier.  This ensures that links inside the element create *informative* index refs, not normative ones.

* `<informative>`
* `<inf>`
* `<non-normative>`
* something else?

**Comment by domenic** [comment]
How do you currently mark sections as informative?

**Comment by tabatkins** [comment]
A "note", "example", or "non-normative" class on an ancestor..

**Comment by domenic** [comment]
Yeah a shorthand would be useful especially if it means I don't have to end up with `<div class="non-normative">` in the output when I'm only using it to tell Bikeshed meta-information.

**Comment by Quuxplusone** [comment]
Maybe I misunderstand the purpose of "normative index refs" and "informative index refs."  I thought the normativeness of the index entry was a property of the *referenced document*, not a property of the *context in which the reference was made.*

So for example if I had a non-normative note that said `Note: For background information on foobar, see ISO/IEC JTC1/SC22/WG21 [[!N4750]]`, I've always assumed I'd want N4750 to appear in the *normative* index, because it's a reference *to* a normative document, even though the reference happens to have been in a non-normative context. (And vice versa, if some normative text referred to a non-normative document... well, that really should never happen, I guess.)

Have I been doing this wrong?

If I have been doing this wrong, then: If N4750 is referenced from two places in my paper — once from a normative paragraph and once from a non-normative paragraph — would I want its entry to show up in *both* indices?

**Comment by tabatkins** [comment]
Yeah, you've been doing it wrong. ^_^ A normative reference is one that your document *normatively depends on* - some part of your document depends on a thing defined in this other document, and won't work correctly without it. An informative reference is just pointing to something that can be useful to read for some reason, but that isn't required for implementing your document.  (I use informative references to point to other specs doing a particular thing that's similar to the thing I'm currently defining, for example.)

All your normative references will be, by definition, only to normative documents, but your informative references can be to anything.

To avoid misleading people about the set of documents that your document depends on, if something is referred to both normatively and informatively, it only shows up in the normative appendix.

Apologies for this not being obvious; I'm steeped enough in W3C-ese that this didn't register as something that needed explaining. ^_^ I'll fix the docs to make this clearer.

**Comment by Quuxplusone** [comment]
> A normative reference is one that your document normatively depends on - some part of your document depends on a thing defined in this other document, and won't work correctly without it. An informative reference is just pointing to something that can be useful to read for some reason, but that isn't required for implementing your document.

Well! I'll be updating some of my drafts. :)

**Comment by svgeesus** [comment]
It would be helpful if bikeshed output `section` elements; at minimum, for the normative and informative reference sections. Currently, even though a human can tell from the headings, tools have problems and thus erroneously report normative references that are in fact informative.
